lighthouse-viewer/gulpfile.js

```diff
@@ -40,118 +57,113 @@ gulp.task('lint', () => {
 
 gulp.task('images', () => {
   return gulp.src('app/images/**/*')
-  .pipe(gulp.dest(`${config.dist}/images`));
+  .pipe(gulp.dest(`dist/images`));
 });
 
 gulp.task('concat-css', () => {
-  return gulp.src([
-    '../lighthouse-core/report/styles/report.css',
-    '../lighthouse-core/report/partials/*.css',
-    'app/styles/viewer.css',
-  ])
-  .pipe($.concat('viewer.css'))
-  .pipe(gulp.dest(`${config.dist}/styles`));
+  const reportCss = streamFromString(ReportGenerator.reportCss, 'report-styles.css');
+  const viewerCss = gulp.src('app/styles/viewer.css');
+
+  return streamqueue({objectMode: true}, reportCss, viewerCss)
+    .pipe($.concat('viewer.css'))
+    .pipe(gulp.dest(`dist/styles`));
 });
 
 gulp.task('html', () => {
+  const templatesStr = ReportGenerator.reportTemplates;
+
+  return gulp.src('app/index.html')
+    .pipe($.replace(/%%LIGHTHOUSE_TEMPLATES%%/, _ => templatesStr))
+    .pipe(gulp.dest('dist'));
+});
+
+gulp.task('pwa', () => {
   return gulp.src([
-    'app/*.html',
     'app/sw.js',
-    'app/manifest.json'
-  ]).pipe(gulp.dest(config.dist));
+    'app/manifest.json',
+  ]).pipe(gulp.dest('dist'));
 });
 
 gulp.task('polyfills', () => {
   return gulp.src([
     'node_modules/url-search-params/build/url-search-params.js',
     'node_modules/whatwg-fetch/fetch.js'
   ])
-  .pipe(gulp.dest(`${config.dist}/src/polyfills`));
+  .pipe(gulp.dest(`dist/src/polyfills`));
 });
 
-gulp.task('browserify', () => {
-  return gulp.src([
-    'app/src/main.js'
-  ], {read: false})
-    .pipe($.tap(file => {
-      const bundle = browserify(file.path, {debug: false})
-        .plugin('tsify', { // Note: tsify needs to come before transforms.
-          allowJs: true,
-          target: 'es5',
-          diagnostics: true,
-          pretty: true,
-          removeComments: true
-        })
-        .transform('brfs')
-        .ignore('../lighthouse-logger/index.js')
-        .ignore('whatwg-url')
-        .ignore('url')
-        .ignore('debug/node')
-        .ignore('source-map')
-        .bundle();
-
-      file.contents = bundle; // Inject transformed content back the gulp pipeline.
-    }))
-    .pipe(gulp.dest(`${config.dist}/src`));
-});
-
-gulp.task('compile', ['browserify'], () => {
-  return gulp.src([`${config.dist}/src/main.js`])
-    .pipe($.uglify()) // minify.
-    .pipe(license())  // Add license to top.
-    .pipe(gulp.dest(`${config.dist}/src`));
+gulp.task('compile-js', () => {
+  const filename = __dirname + '/../lighthouse-core/report/v2/report-generator.js';
```

Comment:
> how big is the viewer bundle now? without fb and such.

183KB

(About 100KB of that is the unminified js [pulled in by ReportGenerator as a string](https://github.com/GoogleChrome/lighthouse/blob/2671c00ce6a5671822e20ba24a2fcc932ff3506e/lighthouse-core/report/v2/report-generator.js#L11-L21), which can be brought down to about 25K with a simple uglify step, but we'd need to replace brfs with something that can chain transforms like that...good fodder for a followup)